src/scenes/AcceleratorRouter/Modules/EventsTable.tsx

```diff
@@ -71,48 +71,77 @@ export default function EventsTable(props: EventsTableProps): JSX.Element {
   const [selectedRows, setSelectedRows] = useState<ModuleEventPartial[]>([]);
   const [eventToEdit, setEventToEdit] = useState<ModuleEventPartial>();
 
+  const generatedIds = new Set();
+
+  function generateUniqueId() {
+    let newId: number;
+    do {
+      newId = Math.random();
+    } while (generatedIds.has(newId)); // Check for uniqueness
+
+    generatedIds.add(newId);
+    return newId;
+  }
+
   const onAddEvent = (eventToAdd: ModuleEventPartial) => {
+    let newEventToAdd = { ...eventToAdd };
+    if (eventToAdd.id === -1 && !eventToAdd.feId) {
+      newEventToAdd = { ...eventToAdd, feId: generateUniqueId() };
+    }
     if (setEventsToAdd) {
       setEventsToAdd((prev) => {
         if (prev) {
-          return [...prev, eventToAdd];
+          return [...prev, newEventToAdd];
         } else {
-          return [eventToAdd];
+          return [newEventToAdd];
         }
       });
     }
   };
 
   const onEditedEvent = (editedEvent: ModuleEventPartial) => {
```

Comment:
This function is pretty large. A couple of suggestions:
1. Try to remove unnecessary nesting. For example, this line `if (setEventsToDelete && setEventsToAdd) {` wraps the entire function contents. Instead, you can do something like
```
if (!(setEventsToDelete && setEventsToAdd)) {
  return;
}
```

2. Break out logical branches into their own functions:
For example: 
```
if (editedEvent.id === -1) { 
  addEvent(editedEvent) 
} else {
  editEvent(editedEvent)
}
```

Reply:
Thank for the suggestions!